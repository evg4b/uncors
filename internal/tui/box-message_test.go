package tui_test

import (
	"strings"
	"testing"

	"github.com/evg4b/uncors/testing/testutils"
	"github.com/gkampitakis/go-snaps/snaps"

	"github.com/evg4b/uncors/internal/tui"
)

func TestPrintWarningBox(t *testing.T) {
	tests := []struct {
		name    string
		message string
	}{
		{
			name:    "render single line message",
			message: "test message",
		},
		{
			name:    "render multi line message",
			message: "test message\nsecond line",
		},
	}
	for _, testCase := range tests {
		t.Run(testCase.name, testutils.WithTrueColor(func(t *testing.T) {
			buffer := strings.Builder{}

			tui.PrintWarningBox(&buffer, testCase.message)

			snaps.MatchSnapshot(t, buffer.String())
		}))
	}
}

func TestPrintInfoBox(t *testing.T) {
	tests := []struct {
		name    string
		message string
	}{
		{
			name:    "render single line message",
			message: "test message",
		},
		{
			name:    "render multi line message",
			message: "test message\nsecond line",
		},
	}
	for _, testCase := range tests {
		t.Run(testCase.name, testutils.WithTrueColor(func(t *testing.T) {
			buffer := strings.Builder{}

			tui.PrintInfoBox(&buffer, testCase.message)

			snaps.MatchSnapshot(t, buffer.String())
		}))
	}
}
