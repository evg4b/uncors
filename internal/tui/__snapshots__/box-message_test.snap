
[TestPrintWarningBox/render_single_line_message - 1]
[48;2;255;211;0m [0m[38;2;0;0;0;48;2;255;211;0mWARN [0m[48;2;255;211;0m [0m test message

---

[TestPrintWarningBox/render_multi_line_message - 1]
[48;2;255;211;0m [0m[38;2;0;0;0;48;2;255;211;0mWARN [0m[48;2;255;211;0m [0m test message
[48;2;255;211;0m [0m[38;2;0;0;0;48;2;255;211;0m[0m[48;2;255;211;0m [0m[48;2;255;211;0m     [0m second line 

---

[TestPrintInfoBox/render_single_line_message - 1]
[48;2;0;113;206m [0m[38;2;0;0;0;48;2;0;113;206mINFO [0m[48;2;0;113;206m [0m test message

---

[TestPrintInfoBox/render_multi_line_message - 1]
[48;2;0;113;206m [0m[38;2;0;0;0;48;2;0;113;206mINFO [0m[48;2;0;113;206m [0m test message
[48;2;0;113;206m [0m[38;2;0;0;0;48;2;0;113;206m[0m[48;2;0;113;206m [0m[48;2;0;113;206m     [0m second line 

---
